### ISSUE TITLE
handling errors

### ISSUE THREAD

**Issue by calvinmetcalf** [calvinmetcalf/async-iter-stream#4]
are we handling errors right?

**Comment by mcollina** [comment]
I don't think so. I think we should `.destroy()`  if there is an error somewhere in the for loop. Is this possible?

**Comment by calvinmetcalf** [comment]
yeah we can make it so the `error` method passes the error to destroy
